File: app/src/main/java/com/varunbarad/popularmovies/util/YouTubeApiHelper.java
package com.varunbarad.popularmovies.util;

import java.util.Locale;

/**
 * Creator: Varun Barad
 * Date: 23-10-2017
 * Project: PopularMovies
 */
public final class YouTubeApiHelper {
  private static final String THUMBNAIL_IMAGE_FORMAT = "https://i.ytimg.com/vi/%s/default.jpg";
  
  public static boolean isValidVideoId(String videoId) {
    return videoId.trim().matches("[0-9a-zA-Z-_]{11}");
  }
  
  public static String getDefaultThumbnailUrl(String videoId) {
    if (YouTubeApiHelper.isValidVideoId(videoId)) {
      return String.format(Locale.ENGLISH, THUMBNAIL_IMAGE_FORMAT, videoId);
    } else {
      return null;
    }
  }
}

File: app/src/main/java/com/varunbarad/popularmovies/util/YouTubeApiHelper.kt
@file:JvmName("YouTubeApiHelper")
package com.varunbarad.popularmovies.util

/**
 * Creator: Varun Barad
 * Date: 2019-06-03
 * Project: PopularMovies
 */

fun String.isValidVideoId(): Boolean {
    return this.trim().matches(Regex("[0-9a-zA-Z-_]{11}"))
}

fun getDefaultThumbnailUrl(videoId: String): String? {
    return if (videoId.isValidVideoId()) {
        "https://i.ytimg.com/vi/$videoId/default.jpg"
    } else {
        null
    }
}
